pkg/services/contexthandler/contexthandler.go

```diff
@@ -312,7 +312,8 @@ func (h *ContextHandler) initContextWithAPIKey(reqContext *models.ReqContext) bo
 	}
 
 	if apikey.ServiceAccountId == nil || *apikey.ServiceAccountId < 1 { //There is no service account attached to the apikey
-		//Use the old APIkey method.  This provides backwards compatibility.
+		// Use the old APIkey method.  This provides backwards compatibility.
+		// will probably have to be supported for a long time.
```

Comment:
> // will probably have to be supported for a long time.

i don't have a lot of context here, but are there downsides to making this migration automatic?
<img width="1397" alt="image" src="https://user-images.githubusercontent.com/360020/199482787-d2617588-32f8-4d89-8d5a-46e1b9858828.png">

Reply:
Or perhaps automatic if no keys exist

Reply:
this is a good take. I'll add this to our backlog

Reply:
@toddtreece alot of our customers rely on apikeys and we want to make sure that they have been aware of the change to service accounts.

Are you referring to a new instance in general, where they do not have any apikeys and we set service accounts tab only by default? 🤔

Reply:
if it's the latter case, I totally agree that that does make sense ⚡

Reply:
@eleijonmarck i guess i meant either way. i have no context about what actually happens behind the scenes when an API key is migrated using the `Migrate to service accounts now` button, i just see that a service account has been created and the API key still works as it did before migration:
![image](https://user-images.githubusercontent.com/360020/199575134-f3e8cdfb-88ca-4f32-81b1-a0054616a8f5.png)

and then the api keys have this notice:
![image](https://user-images.githubusercontent.com/360020/199576532-a5b36bff-13dc-4413-b96c-e62e05edeb12.png)

Reply:
added a new issue to the backlog - https://github.com/grafana/grafana/issues/58091

Reply:
@toddtreece this is exactly the intended experience. We were the ones developing this workflow, so any feedback is welcome.
We could add some more detail about what is meant by migrating from apikey to service accounts.

Reply:
nope, no issues with the workflow for migration, it works great! the `// will probably have to be supported for a long time` comment made me wonder if API keys could be automatically migrated at some point so that the logic for the old keys can be removed. a bit of a tangent since i was just curious about the plan, but that can be discussed outside of this PR :smile: